src/Xamarin.Android.Build.Tasks/Tasks/CollectNonEmptyDirectories.cs

```diff
@@ -22,7 +22,8 @@ public override bool Execute ()
 				var firstFile = Directory.EnumerateFiles(directory.ItemSpec, "*.*", SearchOption.AllDirectories).FirstOrDefault ();
 				if (firstFile != null) {
 					var taskItem = new TaskItem (directory.ItemSpec, new Dictionary<string, string> () {
-						{"FileFound", firstFile}
+						{"FileFound", firstFile },
+						{"StampFile", Path.GetFullPath (Path.Combine (directory.ItemSpec, "..", "..")) + ".stamp" },
```

Comment:
This part scares me: *is* `Path.Combine (directory.ItemSpec, "..", "..")` a valid & writable directory?

So I look at the call-site:

```xml
	<CollectNonEmptyDirectories Directories="@(LibraryResourceDirectories);@(_AdditonalAndroidResourceCachePaths->'%(Identity)\res')">
		<Output TaskParameter="Output" ItemName="_LibraryResourceDirectories" />
	</CollectNonEmptyDirectories>
```

Inputs are `@(LibraryResourceDirectories)` and `@(_AdditonalAndroidResourceCachePaths)`.

`@(_AdditonalAndroidResourceCachePaths)` looks safe, as that comes from `<CalculateAdditionalResourceCacheDirectories/>`, which constructs filenames "underneath" `CalculateAdditionalResourceCacheDirectories.CacheDirectory`, which is `$(MonoAndroidIntermediateResourceCache)`, so that should be writable.

But where does `@(LibraryResourceDirectories)` come from?  I don't see what sets it, and thus I can't ascertain that the directory is actually writable.

Plus, it's a "public" property (no `_` prefix), implying someone could do:

```xml
<ItemGroup>
  <LibraryResourceDirectories Include="C:\" />
</ItemGroup>
```

which *has* no `..\..` directory.

Is this actually correct?

Reply:
I think `LibraryResourceDirectories` should not really be public. I suspect that is a artefact of some old code. Users should NOT be setting it.

Reply:
`@(LibraryResourceDirectories)` is from `<ReadLibraryProjectImportsCache/>`:

```xml
	<ReadLibraryProjectImportsCache
		CacheFile="$(_AndroidLibraryProjectImportsCache)">
		<Output TaskParameter="ResolvedResourceDirectories" ItemName="LibraryResourceDirectories" />
```

`ReadLibraryProjectImportsCache.ResolvedResourceDirectories` is/should be equivalent to `ResolveLibraryProjectImports.ResolvedResourceDirectories`, which comes from `ResolveLibraryProjectImports.Assemblies`/etc..

`ResolveLibraryProjectImports.ResolvedResourceDirectoryStamps` also constructs a `../../.stamp` path.

This looks fine then.